code/modules/supply/supply_console.dm

```diff
@@ -0,0 +1,499 @@
+		var/list/order_data = list(
+			"ordernum" = order.ordernum,
+			"supply_type" = order.object.name,
+			"orderedby" = order.orderedby,
+			"comment" = order.comment
+		)
+		orders += list(order_data)
+	return orders
+
+/obj/machinery/computer/supplycomp/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	if(!cargo_account || !account_database)
+		reconnect_database()
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "CargoConsole", name, 900, 800, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/supplycomp/ui_data(mob/user)
+	var/list/data = list()
+
+	data["requests"] = build_request_data(user)
+	data["accounts"] = build_account_data(user)
+	data["orders"] = build_shopping_list_data()
+	data["shipments"] = build_shipment_list_data()
+	data["is_public"] = is_public
+	data["moving"] = SSshuttle.supply.mode != SHUTTLE_IDLE
+	data["at_station"] = SSshuttle.supply.getDockedId() == "supply_home"
+	data["timeleft"] = SSshuttle.supply.timeLeft(60 SECONDS)
+	data["can_launch"] = !SSshuttle.supply.canMove()
+
+	return data
+
+/obj/machinery/computer/supplycomp/ui_static_data(mob/user)
+	var/list/static_data = list()
+	var/list/packs_list = list()
+
+	for(var/set_name in SSeconomy.supply_packs)
+		var/datum/supply_packs/pack = SSeconomy.supply_packs[set_name]
+		if((pack.hidden && hacked) || (pack.contraband && can_order_contraband) || (pack.special && pack.special_enabled) || (!pack.contraband && !pack.hidden && !pack.special))
+			packs_list.Add(list(list(
+				"name" = pack.name,
+				"cost" = pack.cost,
+				"ref" = "[pack.UID()]",
+				"contents" = pack.ui_manifest,
+				"cat" = pack.group)))
+
+	static_data["supply_packs"] = packs_list
+
+	var/list/categories = list()
+	for(var/category in SSeconomy.all_supply_groups)
+		categories.Add(list(list("name" = get_supply_group_name(category), "category" = category)))
+
+	static_data["categories"] = categories
+
+	return static_data
+
+
+
+/obj/machinery/computer/supplycomp/ui_act(action, list/params, datum/tgui/ui)
+	if(..())
+		return
+
+	var/mob/user = ui.user
+
+	// If its not a public console, and they aint authed, dont let them use this
+	if(!is_public && !is_authorized(user))
+		return
+
+	if(!SSshuttle)
+		stack_trace("The SSshuttle controller datum is missing somehow.")
+		return
+
+	. = TRUE
+	add_fingerprint(user)
+
+	switch(action)
+		if("moveShuttle")
+			move_shuttle(user)
+		if("order")
+			if(world.time < reqtime)
+				visible_message("<b>[src]</b>'s monitor flashes, \"[world.time - reqtime] seconds remaining until another requisition form may be printed.\"")
+				return
+			var/amount = 1
+			if(params["multiple"] == "1") // 1 is a string here. DO NOT MAKE THIS A BOOLEAN YOU DORK
+				var/num_input = input(user, "Amount", "How many crates? (20 Max)") as null|num
```

Comment:
here is says 20. Yet below you clamp it to 10. Can you make a define or variable for this to avoid these issues?

Reply:
added define in next commit